pkg/pillar/cmd/zedagent/handlemetrics.go

```diff
@@ -383,62 +417,40 @@ func publishMetrics(ctx *zedagentContext, iteration int) {
 			&metric)
 	}
 
-	disks := diskmetrics.FindDisksPartitions(log)
-	for _, d := range disks {
-		size, _ := diskmetrics.PartitionSize(log, d)
-		log.Debugf("Disk/partition %s size %d", d, size)
-		size = utils.RoundToMbytes(size)
-		metric := metrics.DiskMetric{Disk: d, Total: size}
-		stat, err := disk.IOCounters(d)
-		if err == nil {
-			metric.ReadBytes = utils.RoundToMbytes(stat[d].ReadBytes)
-			metric.WriteBytes = utils.RoundToMbytes(stat[d].WriteBytes)
-			metric.ReadCount = stat[d].ReadCount
-			metric.WriteCount = stat[d].WriteCount
-		}
-		// XXX do we have a mountpath? Combine with paths below if same?
-		ReportDeviceMetric.Disk = append(ReportDeviceMetric.Disk, &metric)
-	}
-
 	var persistUsage uint64
-	for _, path := range reportDiskPaths {
-		u, err := disk.Usage(path)
-		if err != nil {
-			// Happens e.g., if we don't have a /persist
-			log.Errorf("disk.Usage: %s", err)
-			continue
-		}
-		// We can not run diskmetrics.SizeFromDir("/persist") below in reportDirPaths, get the usage
-		// data here for persistUsage
-		if path == types.PersistDir {
-			persistUsage = u.Used
+	for _, diskMetric := range getAllDiskMetrics(ctx) {
+		var diskPath, mountPath string
+		if diskMetric.IsDir {
+			mountPath = diskMetric.DiskPath
+		} else {
+			diskPath = diskMetric.DiskPath
 		}
-		log.Debugf("Path %s total %d used %d free %d",
-			path, u.Total, u.Used, u.Free)
-		metric := metrics.DiskMetric{MountPath: path,
-			Total: utils.RoundToMbytes(u.Total),
-			Used:  utils.RoundToMbytes(u.Used),
-			Free:  utils.RoundToMbytes(u.Free),
+		metric := metrics.DiskMetric{
+			Disk:       diskPath,
+			MountPath:  mountPath,
+			ReadBytes:  diskMetric.ReadBytes,
```

Comment:
Somewhere (here or in zedagent) this needs to be converted/rounded to Mbytes since that is what should be iin the protobuf message.

Reply:
Changed.